### PR TITLE
Load fixtures from source

### PR DESCRIPTION
**Pull request by Arminek**

### CODE REVIEW

File: src/ApiTestCase.php [hunk 1]
```diff
@@ -228,37 +214,100 @@ private function getCalledClassFolder()
...
 
-        $this->folderExists($calledClassFolder);
+        $this->directoryCheck($calledClassFolder);
 
         return $calledClassFolder;
     }
 
+    protected function loadAllFixtures()
+    {
+        $this->loadAllFixtures();
```
Comment (michalmarcinkowski) [inline src/ApiTestCase.php]: What is the point of this function?

File: src/ApiTestCase.php [hunk 2]
```diff
@@ -228,37 +214,100 @@ private function getCalledClassFolder()
...
+     * @param string $directory
+     *
+     * @return array
+     */
+    private function loadFileNamesFromDirectory($directory)
+    {
+        $tree = scandir($directory);
+        $tree = array_diff($tree, array('.', '..'));
+
+        if (!$tree) {
```
Comment (michalmarcinkowski) [inline src/ApiTestCase.php]: `empty($tree)` will be more explicit.

File: src/ApiTestCase.php (hunk 1 above)
Comment (Arminek) [inline src/ApiTestCase.php]: I know that nothing special happening here but i thought it increase developer experiance.

Comment (pamil) [inline src/ApiTestCase.php]: Either I'm missing something or it makes an infinite loop :)

File: src/ApiTestCase.php [hunk 3]
```diff
@@ -228,37 +214,100 @@ private function getCalledClassFolder()
...
+    }
+
     /**
      * @param string $directory
+     * @param array $fixtureNames
      *
      * @return array
      */
-    private function getFilesFromDirectory($directory)
+    private function getFixturePathsFromDirectory($directory, array $fixtureNames = array())
```
Comment (pamil) [inline src/ApiTestCase.php]: We may use `\RecursiveDirectoryInterator`, it'll make it cleaner and (most probably) faster (and will result in an iterator of handy `\SplFileInfo`).

File: src/ApiTestCase.php (hunk 1 above)
Comment (Arminek) [inline src/ApiTestCase.php]: It should call ``->loadFixturesByName();``

File: src/ApiTestCase.php [hunk 4]
```diff
@@ -228,37 +214,100 @@ private function getCalledClassFolder()
...
     }
 
     /**
      * @param string $directory
      */
-    private function folderExists($directory)
+    private function directoryCheck($directory)
     {
         if (!file_exists($directory)) {
             throw new \RuntimeException(sprintf('Folder %s does not exist', $directory));
```
Comment (pamil) [inline src/ApiTestCase.php]: To be more consistent: `Folder` -> `Directory`?

File: src/ApiTestCase.php [hunk 5]
```diff
@@ -228,37 +214,100 @@ private function getCalledClassFolder()
...
+
+    /**
+     * @param string $name
+     * @param array $fixtureNames
+     *
+     * @return bool
+     */
+    private function isExpectedFixture($name, array $fixtureNames = array())
+    {
+        if (empty($fixtureNames) || in_array($name, $fixtureNames)) {
```
Comment (michalmarcinkowski) [inline src/ApiTestCase.php]: `return (empty($fixtureNames) || in_array($name, $fixtureNames);`

Comment (pamil) [inline src/ApiTestCase.php]: Use `true` as third argument of `in_array` to use strict comparison, loose one won't work as you expect :)

File: src/ApiTestCase.php (hunk 3 above)
Comment (pamil) [inline src/ApiTestCase.php]: This applies to `loadFileNamesFromDirectory($directory)`.

File: src/ApiTestCase.php [hunk 6]
```diff
@@ -228,37 +214,100 @@ private function getCalledClassFolder()
...
+        }
+
+        return false;
     }
 
     /**
      * @param string $directory
      */
-    private function folderExists($directory)
+    private function directoryCheck($directory)
```
Comment (pamil) [inline src/ApiTestCase.php]: `assertDirectoryExists` would be much more verbose :)

File: test/src/Tests/Controller/SampleControllerXmlTest.php [hunk 7]
```diff
@@ -61,6 +61,9 @@ public function testGetResponseFromMockedService()
 
     public function testProductIndexResponse()
     {
+        $this->purgeDataBase();
```
Comment (pamil) [inline test/src/Tests/Controller/SampleControllerXmlTest.php]: `DataBase` -> `Database`

**Comment by pjedrzejewski** [comment]
Looks good to me. :+1: @lchrusciel your call!

File: src/ApiTestCase.php [hunk 8]
```diff
@@ -228,37 +214,91 @@ private function getCalledClassFolder()
...
 
-        $files = scandir($directory);
-        $files = array_diff($files, array('..', '.'));
-        if (!$files) {
-            throw new \RuntimeException(sprintf('There is no files to load in folder %s', $directory));
+        return $this->getFixturePathsFromDirectory($source);
+    }
+
+    /**
+     * @param  $directory
```
Comment (lchrusciel) [inline src/ApiTestCase.php]: missing typehint.

File: src/ApiTestCase.php [hunk 9]
```diff
@@ -228,37 +214,91 @@ private function getCalledClassFolder()
...
+        $this->assertSourceExists($calledClassFolder);
 
         return $calledClassFolder;
     }
 
     /**
-     * @param string $directory
+     * @param string $source
+     */
+    protected function loadFixturesFrom($source)
```
Comment (lchrusciel) [inline src/ApiTestCase.php]: This function should be placed before private ones.

File: src/ApiTestCase.php [hunk 10]
```diff
@@ -228,37 +214,91 @@ private function getCalledClassFolder()
...
+        $this->entityManager->flush();
+    }
+
+    /**
+     * @param string $source
      *
      * @return array
      */
-    private function getFilesFromDirectory($directory)
+    private function getFixturePathsFromSource($source)
```
Comment (lchrusciel) [inline src/ApiTestCase.php]: Won't `getFixturePaths` be enough?

File: test/src/Entity/Product.php [hunk 11]
```diff
@@ -1,9 +1,9 @@
 <?php
 
 /*
- * This file is part of the Sylius package.
+ * This file is part of the ApiTestCase package.
```
Comment (lchrusciel) [inline test/src/Entity/Product.php]: :+1:

**Comment by lchrusciel** [comment]
Good for me. Waiting for green build and it si ready for merge.

**Pull request merged**